### PR TITLE
Display extra pupilage details

### PR DESCRIPTION
**Pull request by tomlovesgithub**
## What's included?
More logic in the application view, presenting the more involved answers for barrister qualifications.

## Who should test?
✅ Product owner
✅ Developers
✅ UTG

## How to test?
Ensure changes made on apply side [see apply #747 for more details] show in application view.

## Risk - how likely is this to impact other areas?
🟠 Medium risk - this does change code that is shared with other areas

## Additional context
image
image
image
---
PREVIEW:DEVELOP
_can be OFF, DEVELOP or STAGING_

### CODE REVIEW

**Review by HalcyonJAC — APPROVED** [review]
Looks good to me :)

File: src/views/Exercises/Applications/Application.vue [hunk 1]
```diff
@@ -995,8 +996,40 @@
...
+                      </dt>
+                      <dd class="govuk-summary-list__value">
+                        <ul class="govuk-list">
+                          <li> {{ item.calledToBarDate | formatDate }}</li>
+                        </ul>
+                      </dd>
+                    </div>
+
+                    <div
+                      v-if="item.notCalledToBar && item.notCalledToBar === true"
```
Comment (warrensearle) [inline src/views/Exercises/Applications/Application.vue:1016]: What is the benefit of having both these checks within the template?
 - `item.notCalledToBar` is truthy
 - `item.notCalledToBar` is true

Wouldn't `v-if="item.notCalledToBar"` do the same... or am I missing something?

**Review by warrensearle — COMMENTED** [review]
See question in comments

File: src/views/Exercises/Applications/Application.vue (hunk 1 above)
Comment (tomlovesgithub) [reply src/views/Exercises/Applications/Application.vue:1016]: right you are!

**Review by mbrookeswebdev — APPROVED** [review]

**Review by warrensearle — APPROVED** [review]
Looks good to me 👍

**Review by KateMJAC — CHANGES_REQUESTED** [review]
Hello Tom, please find the information below regarding the qualification requirements to become a barrister:
So key date for us is the date they complete pupillage – that is when they qualify. But if they have not been required to undertake pupillage, it is the date they are called to the Bar. Section 51(6) of the TCE Act 2007 states: 

For the purposes of section 50 and this section, a person shall be taken first to become a barrister—
(a)when the person completes pupillage in connection with becoming a barrister, or
(b)in the case of a person not required to undertake pupillage in connection with becoming a barrister, when the person is called to the Bar of England and Wales.

Can you please confirm the informaton that informed the decision to include the 'I have not been called to the bar' check box?

**Comment by tomlovesgithub** [comment]
Hi Kate,
re: the pupilage stuff, i understand that the required info is:
Date completed pupilage
OR
Date called to bar
The question here really is surrounding the kinds of validation and minimum satisfactory answer for the qualifications section of the application.
In building this out as i have done, currently the page cannot be saved or progressed until a date has been supplied for one or the other of the above mentioned dates.
However to facilitate any ‘unhappy’ or ‘edge case’ scenarios where for any reason a candidate cannot provide these dates, I’ve allowed this to be bypassed with the checkboxes, with the hopes candidates will be able to provide justification/further information in their details box.
I understand that 99% of candidates who dont have these dates would be disqualified as ineligible, however it seemed counterintuitive to build a dead end into the application, which could force the candidate to provide [inaccurate] information in order to be able to progress. 

- copied from slack

**Comment by KateMJAC** [comment]
@tomlovesgithub all works fine on apply, however there are some fields/rows on admin that appear in certain scenarios when they shouldn't/don't need to. 

- If a candidate selects 'yes' to them completing pupillage and provided their dates, admin does not need to display the following: 'Called to the bar     No'.

The act of confirming 'yes' to completing pupillage and providing a date indicates that they weren't called to the bar, 

- If a candodate selects 'yes' on 'have you been called to the bar?' and provides a date, we do not need the following row: 'Did not complete pupillage notes'.

The didn't complete pupillage text field isn't on Apply if a candidate confirms they were called to the bar and provides a date, as no further information is required.

If you can remove these fields/rows from admin, I can then send it out to the UTG.

Thank you

**Review by KateMJAC — DISMISSED** [review]
@tomlovesgithub all works fine on apply, however there are some fields/rows on admin that appear in certain scenarios when they shouldn't/don't need to. 

- If a candidate selects 'yes' to them completing pupillage and provided their dates, admin does not need to display the following: 'Called to the bar     No'.

The act of confirming 'yes' to completing pupillage and providing a date indicates that they weren't called to the bar, 

- If a candodate selects 'yes' on 'have you been called to the bar?' and provides a date, we do not need the following row: 'Did not complete pupillage notes'.

The didn't complete pupillage text field isn't on Apply if a candidate confirms they were called to the bar and provides a date, as no further information is required.

If you can remove these fields/rows from admin, I can then send it out to the UTG.

Thank you

**Review by KateMJAC — COMMENTED** [review]
@tomlovesgithub the above issues still appear on Admin.

**Comment by joy-ade** [comment]
@tomlovesgithub Positive test from UTG, however we receive the following feedback in relation to upload section.

[UserTest4.docx](https://github.com/jac-uk/admin/files/6607853/UserTest4.docx)

**Comment by KateMJAC** [comment]
Awaiting response from @tomlovesgithub before this ticket can be progressed.

**Comment by KateMJAC** [comment]
This has been sent to UTG on 30/06/21 after final changes made by Tom. Will await feedback before final approval. 

Also before this ticket is released consideration needs to be given to how it may/may not impact open exercises.

**Review by KateMJAC — COMMENTED** [review]
This has been sent to UTG on 30/06/21 after final changes made by Tom. Will await feedback before final approval. 

Also before this ticket is released consideration needs to be given to how it may/may not impact open exercises.

**Review by KateMJAC — APPROVED** [review]
Positive result at UTG.

**Pull request merged**